packages/wallet/src/redux/protocols/concluding/states.ts

```diff
@@ -0,0 +1,109 @@
+import { Properties as P } from '../../utils';
+
+export type ConcludingState = NonTerminalState | TerminalState;
+export type ConcludingStateType = ConcludingState['type'];
+
+export type NonTerminalState =
+  | AcknowledgeConcludingImpossible
+  | ApproveConcluding
+  | WaitForOpponentConclude
+  | AcknowledgeChannelConcluded
+  | WaitForDefund;
+
+export type TerminalState = Success | Failure;
+
+export type FailureReason =
+  | 'NotYourTurn'
+  | 'ChannelDoesntExist'
+  | 'ConcludeCancelled'
+  | 'DefundFailed';
+
+export interface AcknowledgeConcludingImpossible {
+  type: 'AcknowledgeConcludingImpossible';
+  processId: string;
+}
+export interface ApproveConcluding {
+  type: 'ApproveConcluding';
+  processId: string;
+}
+
+export interface WaitForOpponentConclude {
+  type: 'WaitForOpponentConclude';
+  processId: string;
+}
+
+export interface AcknowledgeChannelConcluded {
+  type: 'AcknowledgeChannelConcluded';
+  processId: string;
+}
+
+export interface WaitForDefund {
+  type: 'WaitForDefund';
+  processId: string;
+}
+
+export interface Failure {
+  type: 'Failure';
+  reason: FailureReason;
+}
+
+export interface Success {
+  type: 'Success';
+}
+
+// -------
+// Helpers
+// -------
+
+export function isTerminal(state: ConcludingState): state is Failure | Success {
+  return state.type === 'Failure' || state.type === 'Success';
+}
+
+export function isSuccess(state: ConcludingState): state is Success {
+  return state.type === 'Success';
+}
+
+export function isFailure(state: ConcludingState): state is Failure {
+  return state.type === 'Failure';
+}
+
+// ------------
+// Constructors
+// ------------
+export function acknowledgeConcludingImpossible(
+  p: P<AcknowledgeConcludingImpossible>,
+): AcknowledgeConcludingImpossible {
+  const { processId } = p;
+  return { type: 'AcknowledgeConcludingImpossible', processId };
+}
+
+export function approveConcluding(p: P<ApproveConcluding>): ApproveConcluding {
```

Comment:
Could use the `Constructor` helper here instead of `P`.

Reply:
Done.